src/mmrelay/meshtastic_utils.py

```
@@ -250,6 +261,306 @@ def _get_metadata_executor() -> ThreadPoolExecutor:
+    For `get_device_metadata_request`, Meshtastic responses may be:
+    - A ROUTING_APP ACK/NAK packet with `decoded.routing.errorReason`, or
+    - An ADMIN_APP response packet that includes `requestId` but no
+      `decoded.routing`.
+
+    We treat either as forward progress for liveness probing by setting the same
+    acknowledgment flags used by `waitForAckNak()`.
+    """
+    iface = getattr(local_node, "iface", None)
+    ack_state = getattr(iface, "_acknowledgment", None)
+    if ack_state is None:
+        raise RuntimeError("Meshtastic local node missing acknowledgment state")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)